docs/en/table_design/StarRocks_table_design.md

```diff
@@ -2,74 +2,134 @@
+#### Bucketing
 
-### Per-column index
+The second level is bucketing: Data within a partition is further divided into smaller data management units through bucketing. Replicas of each bucket are evenly distributed across BEs to ensure high data availability.
 
-StarRocks supports per-column indexes such as Bloom filters, zone maps, and bitmap indexes:
+StarRocks provides two bucketing methods:
 
-- A Bloom filter is used to determine whether data blocks contain the values you want to query.
+- Hash bucketing: Data is distributed into buckets based on the hash values of the bucketing key. When the bucketing key consists of columns that are frequently used as the conditional columns in queries, query efficiency can be greatly improved.
+- Random bucketing: Data is randomly distributed to buckets. This bucketing method is more simple and ease to use.
 
-- A zone map is used to locate the values within a specified range.
+### [Data types](../sql-reference/sql-statements/data-types/data-type-list.md)
 
-- A bitmap index is used to locate the rows that meet specified query conditions in a column of the ENUM data type.
+In addition to basic data types such as NUMERIC, DATE, and STRING, StarRocks supports complex semi-structured data types, including ARRAY, JSON, MAP, and STRUCT.
+
+### Index
+
+An index is a special data structure and is used as a pointer to data in a table. When the conditional columns in queries are indexed columns, StarRocks can swiftly locate the data that meets the conditions.
+
+StarRocks provides a built-in primary index, which is a prefix index composed of the prefixes of the table's sort key columns. StarRocks also allows you to create secondary indexes, including Bitmap and Bloom Filter indexes, by hand to further enhance query efficiency, .
+
+### Constraints
+
+Constraints help ensure data integrity, consistency, and accuracy. The primary key columns in Primary Key tables are constrained to have unique and NOT NULL values. The aggregate key columns in Aggregate tables and the unique key columns in Unique Key tables are constrained to have unique values.
+
+### More features
+
+Apart from the above features, you can adopt more features based on your business requirements to design a more robust table structure. For example, using Bitmap and HLL columns to accelerate deduplication counting, specifying generated columns or auto-increment columns to speed up some queries, configuring flexible and automatic mechanism for storage cooldown to reduce maintainance costs, and configuring Colocate Join to speed up multi-table JOIN queries. For more details, see [CREATE TABLE](../sql-reference/sql-statements/data-definition/CREATE_TABLE.md).
```

Comment:
1. deduplication counting --- distinct counting    2. configuring flexible and automatic storage cooldown methods to xxx